src/components/CompactServer/CompactServer.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import {ICompactServerProps, ServerStatus} from './CompactServer.Props';
+import {TagContainer} from '../TagContainer/TagContainer';
+import { Icon } from '../Icon/Icon';
+import  {autobind} from '../../utilities/autobind';
+import'./CompactServer.scss';
+
+export class CompactServer extends React.Component<ICompactServerProps, any> {
+
+    constructor(props?: ICompactServerProps) {
+        super(props);
+    }
+
+    getRoleDisplay(role) {
+        return role.display;
+    }
+    
+     public shouldComponentUpdate(nextProps: ICompactServerProps, nextState) {
+        return !(this.props.serverName === nextProps.serverName
+            && this.props.status === nextProps.status
+            && this.props.roleList.map(this.getRoleDisplay).sort().join(',') === nextProps.roleList.map(this.getRoleDisplay).sort().join(',')

```

Comment:
can we make a reference check here as to avoid deep comparing two arrays in should component update?